scipy/signal/_signaltools.py

```diff
@@ -4625,3 +4625,142 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=True):
         sl[axis] = slice(None, None, q)
 
     return y[tuple(sl)]
+
+def envelope(x,N = None,method = 'analytic'):
+    """
+    Calculate the upper and lower envelopes of a 1-D signal.
+
+    By default, the envelopes of `x` are calculated by the magnitude of its
+    analytical signal.
+
+    Parameters
+    ----------
+    x : array_like
+        The signal for which the envelopes are going to be calculated, as a
+        1-dimensional array.
+    N : int, optional
```

Comment:
If you need to have `N` then `N` is not optional. Either needs a default value or must be required. Then you can ignore if not needed inside the body of the function. Usability is the top priority not the algorithm in the API choices.